docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+    builder.AddBlobServiceClient(storageUrl);
+
+    // Use DefaultAzureCredential by default
+    builder.UseCredential(new DefaultAzureCredential());
+  });
+
+  services.AddControllers();
+}
+```
+
+In code above, you need to explicitly specify the `keyVaultUrl` and `storageUrl`. Both variables are `Uri` types. The [Store configuration separately from code](#store-configuration-separately-from-code) section shows how you can avoid specifying the urls explicitly.
+
+The code above uses `DefaultAzureCredential` for authentication. [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential) will choose the best authentication mechanism based on your environment, allowing you to move your app seamlessly from development to production with no code changes.
+
+## Use the registered clients
+
+With the clients registered in `Startup`, you can now use them:
+
+```csharp
+[ApiController]
+[Route("[controller]")]
+public class MyApiController : ControllerBase
+{
+  private readonly BlobServiceClient blobServiceClient;
+
+  public MyApiController(BlobServiceClient blobServiceClient)
+  {
+    this.blobServiceClient = blobServiceClient;
+  }
+
+  /// Get a list of all the blobs in the demo container
+  [HttpGet]
+  public async Task<IEnumerable<string>> Get()
+  {
+    var containerClient = this.blobServiceClient.GetBlobContainerClient("demo");
+    var results = new List<string>();
+    await foreach (BlobItem blob in containerClient.GetBlobsAsync()) 
+    {
+      results.Add(blob.Name);
+    }
+    return results.ToArray();
+  }
+}
+```
+
+## Store configuration separately from code
+
+Recall in the [Register client](#register-client) section above, you explicitly specify the `keyVaultUrl` and `storageUrl`. This could cause problems when you run code against different environments during development and production. The ASP.NET Core team suggests [storing such configurations in environment dependent JSON files](/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#default-configuration). Thus, you can have an `appsettings.Development.json` file with one set of settings and an `appsettings.Production.json` with another set of configurations. The format of the file is:
+
+```json
+{
+  "AzureDefaults": {
+    "Diagnostics": {
+      "IsTelemetryDisabled": false,
+      "IsLoggingContentEnabled": true
+    },
+    "Retry": {
+      "MaxRetries": 3,
+      "Mode": "Exponential"
+    }
+  },
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net"
+  },
+  "Storage": {
+    "ServiceUri": "https://mydemoaccount.storage.windows.net"
+  }
+}
+```
+
+You can add any options from the [ClientOptions](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/ClientOptions.cs) into the `AzureDefaults` section. One of the options is the retry policy. See the section [Configure a new try policy](#configure-a-new-retry-policy).
```

Comment:
```suggestion
You can add any options from the <xref:Azure.Core.ClientOptions> into the `AzureDefaults` section. One of the options is the retry policy. For more information, see [Configure a new try policy](#configure-a-new-retry-policy).
```

Use an xref instead, which correctly points to the API ref docs. For related content links, the MS guidance is to phrase it as "For more information, see [Title to the thing](the link)."